Cesium3DTilesSelection/src/ImplicitOctreeLoader.cpp

```diff
@@ -380,67 +326,28 @@ CesiumAsync::Future<TileLoadResult> ImplicitOctreeLoader::loadTileContent(
     std::string subtreeUrl =
         resolveUrl(_baseUrl, _subtreeUrlTemplate, subtreeID);
-    std::string tileUrl = resolveUrl(_baseUrl, _contentUrlTemplate, *pOctreeID);
-    CesiumGltf::Ktx2TranscodeTargets ktx2TranscodeTargets =
-        contentOptions.ktx2TranscodeTargets;
-
-    SubtreeContentInitializer subtreeInitializer{this, subtreeID};
     return SubtreeAvailability::loadSubtree(
                3,
                asyncSystem,
                pAssetAccessor,
                pLogger,
                subtreeUrl,
                requestHeaders)
-        .thenInWorkerThread(
-            [pLogger,
-             asyncSystem,
-             pAssetAccessor,
-             tileUrl = std::move(tileUrl),
-             subtreeID,
-             octreeID = *pOctreeID,
-             requestHeaders,
-             ktx2TranscodeTargets,
-             subtreeInitializer = std::move(subtreeInitializer)](
-                std::optional<SubtreeAvailability>&&
-                    subtreeAvailability) mutable {
-              if (subtreeAvailability) {
-                bool tileHasContent = isTileContentAvailable(
-                    subtreeID,
-                    octreeID,
-                    *subtreeAvailability);
-
-                subtreeInitializer.subtreeAvailability =
-                    std::move(*subtreeAvailability);
-
-                // subtree is available, so check if tile has content or not. If
-                // it has, then request it
-                if (!tileHasContent) {
-                  // check if tile has empty content
-                  return asyncSystem.createResolvedFuture(TileLoadResult{
-                      TileEmptyContent{},
-                      TileLoadResultState::Success,
-                      nullptr,
-                      std::move(subtreeInitializer)});
-                }
-
-                return requestTileContent(
-                    pLogger,
-                    asyncSystem,
-                    pAssetAccessor,
-                    tileUrl,
-                    requestHeaders,
-                    ktx2TranscodeTargets,
-                    std::move(subtreeInitializer));
-              }
-
-              return asyncSystem.createResolvedFuture<TileLoadResult>(
-                  TileLoadResult{
-                      TileUnknownContent{},
-                      TileLoadResultState::Failed,
-                      nullptr,
-                      {}});
-            });
+        .thenInMainThread([this, subtreeID](std::optional<SubtreeAvailability>&&
+                                                subtreeAvailability) mutable {
+          if (subtreeAvailability) {
+            this->addSubtreeAvailability(
+                subtreeID,
+                std::move(*subtreeAvailability));
+          }
+
+          // tell client to retry later
+          return TileLoadResult{
+              TileUnknownContent{},
+              TileLoadResultState::RetryLater,
+              nullptr,
+              {}};
```

Comment:
Subtree is now added back to the loader in the main thread as soon as possible. It doesn't need to wait for the tile to be loaded before adding back to the loader